### DISCUSSION TITLE
Convert `qsv enum --hash` output to base62?

### DISCUSSION THREAD

**Discussion by tmtmtmtm** [dathere/qsv#1978] [Q&A] [answered]
I'm finding the new `qsv enum --hash` to be really useful in lots of scenarios, but I'd really like to shrink the resulting values down a bit, and one obvious route is to convert them to base62.

I can do that by piping the output through `perl -pe 's{\d+$}{…}`, but that's fairly gnarly.

My suspicion is that `qsv luau map` would be a much better route here, but despite lots of peering at the existing examples, I'm afraid I still can't get my ahead around how to actually do that. Perhaps this could make a good Cookbook example?

Or is there a much simpler approach that I'm overlooking?

**Comment by jqnatividad** [comment]
@tmtmtmtm , I found this luau library that is perfect for the job - 

https://github.com/Reselim/Base64

I'll whip up an example and add it to the Cookbook.

I'm also thinking of adding base62 encode/decode operations to the `apply` command.

**Reply by tmtmtmtm** [reply]
Hi @jqnatividad — Does that library also handle base62? It looks to me like it's only base64, which would be a lot more awkward in my case (from what I can see it uses + and / for the extra characters which would both cause problems, though I could obviously run the output through `qsv apply replace` for each of those)

For the cookbook, showing how to call out to a library like that would definitely still be super-useful, but if you're going to add something to `apply` maybe it would make sense to add a way to convert between any two bases?

**Reply by jqnatividad** [reply]
Ooops... my bad - lazy googling c&p 😉 

But there is this other snippet for luau that we can possibly use
https://gist.github.com/linsonder6/c87ddf7dc319a5ac5408980556a6f143

For the `apply` command, I'm thinking this fits the bill
https://github.com/fbernier/base62

Also found this reference implementation, though its not a library (yet)
https://github.com/petersmagnusson/base62

WDYT?

**Comment by jqnatividad** [comment, marked as answer]
@tmtmtmtm , I just added the `encode62/decode62` operations to `apply`.  Let me know how it works for you.

https://github.com/jqnatividad/qsv/pull/2013
